File: src/main/java/com/eager/questioncloud/point/entity/ChargePointPaymentEntity.java
package com.eager.questioncloud.point.entity;

import com.eager.questioncloud.point.model.ChargePointPayment;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.LocalDateTime;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Entity
@Table(name = "charge_point_payment")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class ChargePointPaymentEntity {
    @Id
    private String paymentId;

    @Column
    private Long userId;

    @Column
    private String receiptUrl;

    @Column
    private LocalDateTime createdAt;

    @Builder
    public ChargePointPaymentEntity(String paymentId, Long userId, String receiptUrl, LocalDateTime createdAt) {
        this.paymentId = paymentId;
        this.userId = userId;
        this.receiptUrl = receiptUrl;
        this.createdAt = createdAt;
    }

    public ChargePointPayment toModel() {
        return ChargePointPayment.builder()
            .paymentId(paymentId)
            .userId(userId)
            .receiptUrl(receiptUrl)
            .createdAt(createdAt)
            .build();
    }
}

File: qc-domain/qc-point/qc-point-core/src/main/kotlin/com/eager/questioncloud/point/infrastructure/entity/ChargePointPaymentEntity.kt
package com.eager.questioncloud.point.infrastructure.entity

import com.eager.questioncloud.entity.BaseCustomIdEntity
import com.eager.questioncloud.point.domain.ChargePointPayment
import com.eager.questioncloud.point.enums.ChargePointPaymentStatus
import com.eager.questioncloud.point.enums.ChargePointType
import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "charge_point_payment")
class ChargePointPaymentEntity(
    @Id var orderId: String,
    @Column var paymentId: String? = null,
    @Column var userId: Long,
    @Enumerated(EnumType.STRING) @Column var chargePointType: ChargePointType,
    @Enumerated(EnumType.STRING) @Column var chargePointPaymentStatus: ChargePointPaymentStatus,
    @Column var createdAt: LocalDateTime,
    @Column var requestAt: LocalDateTime?,
    isNewEntity: Boolean
) : BaseCustomIdEntity<String>(isNewEntity) {
    fun toModel(): ChargePointPayment {
        return ChargePointPayment(
            orderId,
            paymentId,
            userId,
            chargePointType,
            chargePointPaymentStatus,
            createdAt,
            requestAt
        )
    }

    companion object {
        fun createNewEntity(chargePointPayment: ChargePointPayment): ChargePointPaymentEntity {
            return ChargePointPaymentEntity(
                chargePointPayment.orderId,
                chargePointPayment.paymentId,
                chargePointPayment.userId,
                chargePointPayment.chargePointType,
                chargePointPayment.chargePointPaymentStatus,
                chargePointPayment.createdAt,
                chargePointPayment.requestAt,
                true
            )
        }

        fun fromExisting(chargePointPayment: ChargePointPayment): ChargePointPaymentEntity {
            return ChargePointPaymentEntity(
                chargePointPayment.orderId,
                chargePointPayment.paymentId,
                chargePointPayment.userId,
                chargePointPayment.chargePointType,
                chargePointPayment.chargePointPaymentStatus,
                chargePointPayment.createdAt,
                chargePointPayment.requestAt,
                false
            )
        }
    }

    override fun getId(): String {
        return orderId
    }
}
